DigitalLearningSolutions.Data.Tests/Services/RecommendedLearningServiceTests.cs

```diff
@@ -474,7 +434,83 @@ public async Task
             ).MustNotHaveHappened();
         }
 
+        [Test]
+        public async Task
+            GetRecommendedLearningForSelfAssessment_does_return_resources_with_some_unanswered_optional_competencies()

```

Comment:
Which part of this test defines the question as optional?

Reply:
There is no definition that is is optional, we only care that it has no answers. (it should not be possible to reach here with required competencies with no answers)
I've updated the name of the results setup to GivenSelfAssessmentHasResultsForFirstCompetency